### PR TITLE
before download callback when content is None

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
Addresses #925.

```
<|{None}|file_download|label=do your own download|on_action=download_action|name=a name|>

def download_action(state: State, id: str, action: str, payload: dict):
    # payload.get('args')[0]["name"] => "a name"
    # payload.get('args')[0]["url"] => ""
    download(state, "<file path>", "<file name>", "after_download_action")

def after_download action(state: State, id: str, action: str, payload: dict):
    # invoked after download is terminated
    # payload.get('args')[0]["name"] => "<file name>"
    # payload.get('args')[0]["url"] => <calculated url>
```

### CODE REVIEW

File: gui/src/components/Taipy/FileDownload.tsx [hunk 1]
```diff
@@ -62,13 +62,31 @@ const FileDownload = (props: FileDownloadProps) => {
...
-            runXHR(aRef.current, url, name, onAction ? (() => dispatch(createSendActionNameAction(id, module, onAction, name))) : undefined);
+            if (url) {
+                runXHR(
+                    aRef.current,
+                    url,
+                    name,
+                    onAction ? () => dispatch(createSendActionNameAction(id, module, onAction, {name, url})) : undefined
+                );
+            } else {
+                onAction && dispatch(createSendActionNameAction(id, module, onAction, {name, url}));
```
Comment (FabienLelaquais) [inline gui/src/components/Taipy/FileDownload.tsx]: Isn't url null at this point?

File: gui/src/components/Taipy/FileDownload.tsx [hunk 2]
```diff
@@ -62,13 +62,31 @@ const FileDownload = (props: FileDownloadProps) => {
...
+            }
```
Comment (FabienLelaquais) [inline gui/src/components/Taipy/FileDownload.tsx:89]: Isn't url null at this point?

File: src/taipy/gui/viselements.json [hunk 3]
```diff
@@ -526,8 +526,8 @@
           {
             "name": "on_action",
             "type": "Callback",
-            "doc": "The name of a function that is triggered when the download is initiated.<br/>All the parameters of that function are optional:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>id (optional[str]): the identifier of the button.</li>\n<li>action (optional[str]): the name of the action that provoked the change.</li>\n</ul>",
-            "signature": [["state", "State"], ["id", "str"], ["action", "str"]]
+            "doc": "The name of a function that is triggered when the download is terminated (or on user action if content is None).<br/>All the parameters of that function are optional:\n<ul>\n<li>state (<code>State^</code>): the state instance.</li>\n<li>id (optional[str]): the identifier of the button.</li>\n<li>action (optional[str]): the name of the action that provoked the change.</li>\n<li>payload (dict): the details on this callback's invocation.<br/>\nThis dictionary has the following keys:\n<ul>\n<li>args: List where the first element is an object containing the name property and the url of the content.</li>\n</ul>\n</li>\n</ul>",
```
Comment (FabienLelaquais) [inline src/taipy/gui/viselements.json]: The doc for the payload does not seem to be correct...

**Review by FabienLelaquais — APPROVED** [review]
When you're done considering my minor comments, please create an issue for me documenting that feature.
Thx! and well done

File: gui/src/components/Taipy/FileDownload.tsx (hunk 1 above)
Comment (FredLL-Avaiga) [reply gui/src/components/Taipy/FileDownload.tsx]: No it's ""

File: gui/src/components/Taipy/FileDownload.tsx (hunk 2 above)
Comment (FredLL-Avaiga) [reply gui/src/components/Taipy/FileDownload.tsx:89]: No it's ""

File: src/taipy/gui/viselements.json (hunk 3 above)
Comment (FredLL-Avaiga) [reply src/taipy/gui/viselements.json]: How?

Comment (FredLL-Avaiga) [reply src/taipy/gui/viselements.json]: here is the content (formated but untouched)
```
  The name of a function that is triggered when the download is terminated (or on user action if content is None).
  <br />
  All the parameters of that function are optional:\n
  <ul>\n
    <li>state (<code>State^</code>): the state instance.
    </li>\n
    <li>id (optional[str]): the identifier of the button.
    </li>\n
    <li>action (optional[str]): the name of the action that provoked the change.
    </li>\n
    <li>payload (dict): the details on this callback's invocation.
      <br />\nThis dictionary has the following keys:\n
      <ul>\n
        <li>args: List where the first element is an object containing the name property and the url of the content.
        </li>\n
      </ul>\n
    </li>\n
  </ul>

```

**Review by dinhlongviolin1 — APPROVED** [review]

**Pull request merged**